### PR TITLE
loadUserPlusKeysV2 via offline cache

### PR DESCRIPTION
**Pull request by pzduniak**

### CODE REVIEW

File: go/service/user.go [hunk 1]
```diff
@@ -126,13 +129,22 @@ func (h *UserHandler) LoadUserByName(_ context.Context, arg keybase1.LoadUserByN
...
-	netCtx = libkb.WithLogTag(netCtx, "LUPK2")
-	defer h.G().CTrace(netCtx, fmt.Sprintf("UserHandler#LoadUserPlusKeysV2(%+v)", arg), func() error { return err })()
-	p, err := h.G().GetUPAKLoader().LoadV2WithKID(netCtx, arg.Uid, arg.PollForKID)
-	if p != nil {
-		ret = *p
+func (h *UserHandler) LoadUserPlusKeysV2(ctx context.Context, arg keybase1.LoadUserPlusKeysV2Arg) (ret keybase1.UserPlusKeysV2AllIncarnations, err error) {
+	mctx := libkb.NewMetaContext(ctx, h.G()).WithLogTag("LUPK2")
+	defer mctx.Trace(fmt.Sprintf("UserHandler#LoadUserPlusKeysV2(%+v)", arg), func() error { return err })()
+
+	var retp *keybase1.UserPlusKeysV2AllIncarnations
```
Comment (maxtaco) [inline go/service/user.go]: maybe `retp := &ret` and then....

File: go/service/user.go [hunk 2]
```diff
@@ -126,13 +129,22 @@ func (h *UserHandler) LoadUserByName(_ context.Context, arg keybase1.LoadUserByN
...
+	err = h.service.offlineRPCCache.Serve(mctx, arg.Oa, offline.Version(1), "user.loadUserPlusKeysV2", false, arg, &retp, func(mctx libkb.MetaContext) (interface{}, error) {
+		tmp, err := h.G().GetUPAKLoader().LoadV2WithKID(mctx.Ctx(), arg.Uid, arg.PollForKID)
+		if tmp != nil {
+			retp = tmp
```
Comment (maxtaco) [inline go/service/user.go]: `*retp = *tmp` ?

File: go/service/user.go [hunk 3]
```diff
@@ -126,13 +129,22 @@ func (h *UserHandler) LoadUserByName(_ context.Context, arg keybase1.LoadUserByN
...
+		}
+		return tmp, err
+	})
+	if retp != nil {
```
Comment (maxtaco) [inline go/service/user.go]: then nuke this?

File: go/service/user.go [hunk 4]
```diff
@@ -126,13 +129,22 @@ func (h *UserHandler) LoadUserByName(_ context.Context, arg keybase1.LoadUserByN
...
-	defer h.G().CTrace(netCtx, fmt.Sprintf("UserHandler#LoadUserPlusKeysV2(%+v)", arg), func() error { return err })()
-	p, err := h.G().GetUPAKLoader().LoadV2WithKID(netCtx, arg.Uid, arg.PollForKID)
-	if p != nil {
-		ret = *p
+func (h *UserHandler) LoadUserPlusKeysV2(ctx context.Context, arg keybase1.LoadUserPlusKeysV2Arg) (ret keybase1.UserPlusKeysV2AllIncarnations, err error) {
+	mctx := libkb.NewMetaContext(ctx, h.G()).WithLogTag("LUPK2")
+	defer mctx.Trace(fmt.Sprintf("UserHandler#LoadUserPlusKeysV2(%+v)", arg), func() error { return err })()
+
+	var retp *keybase1.UserPlusKeysV2AllIncarnations
+	err = h.service.offlineRPCCache.Serve(mctx, arg.Oa, offline.Version(1), "user.loadUserPlusKeysV2", false, arg, &retp, func(mctx libkb.MetaContext) (interface{}, error) {
```
Comment (maxtaco) [inline go/service/user.go]: For the cache key here, you should zero out the `pollForKID`.  That's just a way to specify cache freshness requirements, and in the case of offline mode, we're returning arbitrarily stale users.

File: go/service/user.go (hunk 1 above)
Comment (pzduniak) [reply go/service/user.go]: I think I just didn't want to do a copy. Made the changes.

**Pull request merged**